### PR TITLE
[TASK] Upgrade Psalm

### PR DESCRIPTION
**Pull request by oliverklee**
Also regenerate the baseline and fix the issues that cannot be covered
by the baseline.

### CODE REVIEW

**Comment by oliverklee** [comment]
The PHP CS warning confuses me.

**Comment by JakeQZ** [comment]
> 
> 
> The PHP CS warning confuses me.

My guess is that the sniffer does not understand `array<array-key, array<string, string>>` and sees it as simply `array`.  I think previously for these situations we have added a separate `@psalm-return` annotation with that syntax whilst retaining the `@return` annotation with the original syntax.

**Comment by oliverklee** [comment]
Yay, green!

File: src/CssInliner.php [hunk 1]
```diff
@@ -713,6 +713,8 @@ private function getCssSelectorPrecedence(string $selector): int
      *
      * @param string $css CSS with comments removed
      *
+     * @psalm-return array<array-key, array<string, string>> Array of string sub-arrays with the keys
```
Comment (JakeQZ) [inline src/CssInliner.php:716]: I think there should be no description here - the description belongs with `@return` - if the description were to change, these could be liable to become out-of-step.

Comment (oliverklee) [reply src/CssInliner.php:716]: Oh, you're right. I'll fix it.

**Comment by oliverklee** [comment]
Fixed and repushed.

**Pull request merged**